File: 1915-number-of-wonderful-substrings/1915-number-of-wonderful-substrings.cpp
class Solution {
public:
    long long wonderfulSubstrings(string word) {
        unordered_map<int,int> cnt = {{0,1}};
        int mask = 0;
        long long ret = 0;
        for(char c: word) {
            mask ^= 1 << (c-'a');
            ret += cnt[mask];
            for(char i = 'a'; i <= 'j'; i++)
                ret += cnt[mask ^ (1 << (i-'a'))];
            cnt[mask]++;
        }
        return ret;
    }
};

File: 1915-number-of-wonderful-substrings/1915-number-of-wonderful-substrings.java
class Solution {
    public long wonderfulSubstrings(String word) {
        long[] cnt = new long[1024];
        long res=0;
        int msk=0;
        cnt[0]=1;
        for(var ch:word.toCharArray()){
            msk^= 1<<(ch-'a');
            res += cnt[msk];
            for(int i=0; i<10; i++){
                res += cnt[msk ^ (1<<i)];
            }
            cnt[msk]++;
        }
        return res;
    }
}
